core/payment/src/lib.rs

```diff
@@ -48,7 +50,9 @@ impl Service for PaymentService {
 
 impl PaymentService {
     pub async fn gsb<Context: Provider<Self, DbExecutor>>(context: &Context) -> anyhow::Result<()> {
-        let db: DbExecutor = context.component();
+        let db = Data::new(context.clone().component());
+        api::allocations::release_allocations(db.clone(), false).await;
```

Comment:
i think its better to call this in or after `PaymentProcessor::new`

Also this struct already has the db context as `self.db_executor` so no need to clone again